R/guide-legend.r

```diff
@@ -202,10 +202,12 @@ guide_train.legend <- function(guide, scale, aesthetic = NULL) {
     return()
   }
 
-  key <- as.data.frame(
-    setNames(list(scale$map(breaks)), aesthetic %||% scale$aesthetics[1]),
-    stringsAsFactors = FALSE
-  )
+  # in the key data frame, use either the aesthetic provided as
+  # argument to this function or, as a fall back, the first in the vector
+  # of possible aesthetics handled by the scale
+  aes_column_name <- aesthetic %||% scale$aesthetics[1]
+  # need to make a tibble here in case the scale returns a list column
+  key <- tibble(!!aes_column_name := scale$map(breaks))
```

Comment:
Do we use tibbles elsewhere?

@thomasp85 do I remember another internal helper for creating data frames that we could use here?

Reply:
Yes, but it is not in the main branch yet. Right now it doesn't support tidy eval, but this could be added of course (or done outside the constructor)

Reply:
@hadley We do use tibbles currently. I found three examples in the main code base:
https://github.com/tidyverse/ggplot2/blob/4af1337383de41d8f9cf13715d68e7f3a2f40fe4/R/summarise-plot.R#L64
https://github.com/tidyverse/ggplot2/blob/1c09bae2aa24320bcb4891c664cccc16efc86a8a/R/facet-null.r#L33
https://github.com/tidyverse/ggplot2/blob/17b45f906c3a55d187915f4f3010836a804051ae/R/facet-.r#L427